Add get_loops_left() function to Tween

Pennycook opened the pull request:
Implements https://github.com/godotengine/godot-proposals/issues/5141.

Adds a new `get_loops_left()` function to `Tween`, allowing developers to reason about how many times a tweening sequence will repeat and whether to expect `finished` or `loop_finished` as the next signal.

Pennycook commented:
Rebased and force-pushed to restart the checks. They all failed last time, but the failures seemed unrelated to my changes.

Pennycook commented:
Tests are passing now, @KoBeWi.

KoBeWi commented on scene/animation/tween.cpp:
```diff
@@ -228,6 +228,14 @@ Ref<Tween> Tween::set_loops(int p_loops) {
 }
 
+int Tween::get_loops_left() const {
+	if (loops == 0) {
```
```suggestion
	if (loops <= 0) {
```
Values below 0 are accepted in `set_loops()`, but they have the same effect as 0.

KoBeWi commented on scene/animation/tween.cpp:
```diff
@@ -228,6 +228,14 @@ Ref<Tween> Tween::set_loops(int p_loops) {
 
+int Tween::get_loops_left() const {
+	if (loops == 0) {
+		return -1; // Infinite loop
```
```suggestion
		return -1; // Infinite loop.
```

KoBeWi reviewed (approved):
Needs squashing.

Pennycook committed:
Add get_loops_left() function to Tween

Implements godotengine/godot-proposals#5141.

Adds a new get_loops_left() function to Tween, allowing developers to
reason about how many times a tweening sequence will repeat and whether
to expect finished or loop_finished as the next signal.

Co-authored-by: Tomek <kobewi4e@gmail.com>

Pennycook commented:
@KoBeWi Squashed and pushed.

Is there anything else that needs to happen before this can be merged? I'm asking to understand the workflow (and make sure I didn't miss a step), not to try and rush this along.

KoBeWi commented:
No, it's ready to merge now.

YuriSizov commented:
Indeed, it is. Thanks!
